solr/solrj/src/java/org/apache/solr/common/util/JavaBinCodec.java

```diff
@@ -848,9 +848,14 @@ protected Map<Object, Object> newMap(int size) {
     return size < 0 ? new LinkedHashMap<>() : CollectionUtil.newLinkedHashMap(size);
   }
 
-  public Map<Object, Object> readMap(DataInputInputStream dis) throws IOException {
+  public Map<?, Object> readMap(DataInputInputStream dis) throws IOException {
     int sz = readVInt(dis);
-    return readMap(dis, sz);
+
+    if (EnvUtils.getPropertyAsBool("solr.solrj.javabin.mapAsNamedList", true)) {
```

Comment:
Reading this setting on _every single map decode_ is quite bad for performance.  This setting should be a field on the codec, and that which could be toggled with a setter (if useful in a test).